cmd/trace-agent/config/config_test.go

```diff
@@ -1059,6 +1068,390 @@ func TestLoadEnv(t *testing.T) {
 		assert.False(coreconfig.Datadog.GetBool("apm_config.obfuscation.credit_cards.luhn"))
 	})
 
+	t.Run(env, func(t *testing.T) {
+		defer cleanConfig()()
+		assert := assert.New(t)
+		t.Setenv(env, "false")
+		cfg, err := LoadConfigFile("./testdata/full.yaml")
+		assert.NoError(err)
+		assert.False(cfg.Obfuscation.CreditCards.Luhn)
+	})
+
+	env = "DD_APM_OBFUSCATION_ELASTICSEARCH_ENABLED"
+	t.Run(env, func(t *testing.T) {
+		defer cleanConfig()()
+		assert := assert.New(t)
+		t.Setenv(env, "true")
+		_, err := LoadConfigFile("./testdata/full.yaml")
+		assert.NoError(err)
+		assert.True(coreconfig.Datadog.GetBool("apm_config.obfuscation.elasticsearch.enabled"))
+	})
+
+	t.Run(env, func(t *testing.T) {
+		defer cleanConfig()()
+		assert := assert.New(t)
+		t.Setenv(env, "false")
+		cfg, err := LoadConfigFile("./testdata/full.yaml")
+		assert.NoError(err)
+		assert.False(cfg.Obfuscation.ES.Enabled)
+	})
+
+	env = "DD_APM_OBFUSCATION_ELASTICSEARCH_KEEP_VALUES"
+	t.Run(env, func(t *testing.T) {
+		defer cleanConfig()()
+		assert := assert.New(t)
+		t.Setenv(env, `["client_id", "product_id"]`)
+		_, err := LoadConfigFile("./testdata/full.yaml")
+		assert.NoError(err)
+		expected := []string{"client_id", "product_id"}
+		actual := coreconfig.Datadog.GetStringSlice(("apm_config.obfuscation.elasticsearch.keep_values"))
+		if !reflect.DeepEqual(actual, expected) {
+			t.Fatalf("Failed to process env var %s, expected %#v and got %#v", env, expected, actual)
+		}
+	})
+
+	t.Run(env, func(t *testing.T) {
+		defer cleanConfig()()
+		assert := assert.New(t)
+		t.Setenv(env, `["client_id", "product_id"]`)
+		cfg, err := LoadConfigFile("./testdata/full.yaml")
+		assert.NoError(err)
+		expected := []string{"client_id", "product_id"}
+		actual := cfg.Obfuscation.ES.KeepValues
+		if !reflect.DeepEqual(actual, expected) {
+			t.Fatalf("Failed to process env var %s, expected %#v and got %#v", env, expected, actual)
+		}
+	})
```

Comment:
Since the expected value is the same, we can merge the 2 test runs in one (less code duplication):

```golang
	env = "DD_APM_OBFUSCATION_ELASTICSEARCH_KEEP_VALUES"
	t.Run(env, func(t *testing.T) {
		defer cleanConfig()()
		assert := assert.New(t)
		t.Setenv(env, `["client_id", "product_id"]`)
		_, err := LoadConfigFile("./testdata/full.yaml")
		assert.NoError(err)
		expected := []string{"client_id", "product_id"}
		actualGot := coreconfig.Datadog.GetStringSlice(("apm_config.obfuscation.elasticsearch.keep_values"))
                actualParsed := cfg.Obfuscation.ES.KeepValues
                assert.Equal(expected, actualGot)
                assert.Equal(expected, actualParsed)
	})
```

Also in my suggestion I replaced `reflect.DeepEqual` with `assert.Equal` which should be faster. I don't think we need deep equal.

This comment applies to all similar tests below!